onnxruntime/core/providers/nnapi/nnapi_builtin/builders/impl/softmax_op_builder.cc

```diff
@@ -63,16 +63,20 @@ Status SoftMaxOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, cons
   const auto& operand_indices(model_builder.GetOperandIndices());
   const auto& operand_types(model_builder.GetOperandTypes());
   const auto android_feature_level = model_builder.GetEffectiveFeatureLevel();
-  NodeAttrHelper helper(node_unit);
 
-  auto input = node_unit.Inputs()[0].node_arg.Name();
+  // TODO: Needs fix
+  // ??? Not sure why we need this test. At that feature level axis was 1 and input was 2D or 4D.
```

Comment:
I think this check was originally introduced here:

https://github.com/microsoft/onnxruntime/pull/10371/files#diff-1a3b4fc076cd687045e7a8487902619cf1e77805e280cea6f371fa70fb6e6295R1493-R1499

@askhade any chance still recall why do we need this check?

Reply:
I don't see anything in the NNAPI implementation that would make NCHW vs NHWC matter. 

https://android.googlesource.com/platform/frameworks/ml/+/refs/heads/pie-release/nn/common/operations/Activation.cpp#70

It's iterating N and performing a softmax over all the values in the other dims, so whether those are in CHW or HWC order shouldn't matter as that doesn't affect the set of values in each calculation. i.e. there's no way to do a per-channel softmax, so you don't need C to be the second dim.